Add SBD dump gatherer documentation (#137)
trento-project/wanda: guides/gatherers.md

For extra information refer to [trento-project/agent/../gatherers/sbd_test.go](https://github.com/trento-project/agent/blob/main/internal/factsengine/gatherers/sbd_test.go)

### sbd_dump

This gatherer allows accessing the sbd dump command output data.

> It executes the `sbd -d <device> dump` command in all devices configured in the `SBD_DEVICES` field on `/etc/sysconfig/sbd` and aggregates results in only one fact.

Reviewer: Shit, my fault, the entry is `SBD_DEVICE` without the final `s`

### sbd_dump

This gatherer allows accessing the dumped information about all the configured devices on target node.
Simply executes `sbd -d <device> dump` for all the devices in `/etc/sysconfig/sbd` and aggregates results in only one fact.

Note that:
- no arguments are required and any provided argument would just be ignored.
- if dumping any of the devices fails, the fact would be a fact error

```rust
{
  "/path/to/a-device" => {
    "header_version"   => 2.1,
    "number_of_slots"  => 255,
    "sector_size"      => 512,
    "timeout_allocate" => 2,
    "timeout_loop"     => 1,
    "timeout_msgwait"  => 10,
    "timeout_watchdog" => 5,
    "uuid"             => "e09c8993-0cba-438d-a4c3-78e91f58ee52",
  },
  "/path/to/another-device" => {
    "header_version"   => 2.1,
    "number_of_slots"  => 255,
    "sector_size"      => 512,
    "timeout_allocate" => 2,
    "timeout_loop"     => 1,
    "timeout_msgwait"  => 10,
    "timeout_watchdog" => 5,
    "uuid"             => "e5b7c05a-1d3c-43d0-827a-9d4dd05ca54a",
  }
}
```

Specification examples:

```yaml
facts:
  - name: sbd_devices_dump
    gatherer: sbd_dump

  - name: sbd_devices_dump

>     gatherer: whatever_argument # ignored

Reviewer 2: I think you mean here:
```
gatherer: sbd_dump
argument: whatever_argument # ignored
```

Author: Yep thanks!

For extra information refer to [trento-project/agent/../gatherers/sbd_test.go](https://github.com/trento-project/agent/blob/main/internal/factsengine/gatherers/sbd_test.go)

### sbd_dump

> This gatherer allows accessing the dumped data about all the configured devices on target nodes.

Reviewer: I would rephrase this as it is not really accurate:
```
This gatherer allows accessing the sbd dump command output data.

It executes the `sbd -d <device> dump` command in all devices configured in the `SBD_DEVICES` field on `/etc/sysconfig/sbd` and aggregates results in only one fact.
```

> This gatherer allows accessing the dumped information about all the configured devices on target node.

Reviewer 3: ```suggestion
This gatherer allows accessing the dumped data about all the configured devices on target nodes.
```

Author: thanks, changed!

> Simply executes `sbd -d <device> dump` for all the devices in `/etc/sysconfig/sbd` and aggregates results in only one fact.

Reviewer 3: ```suggestion
It executes `sbd -d <device> dump` for all the devices in `/etc/sysconfig/sbd` and aggregates results in only one fact.
```

Author: thanks, changed!

Note that:

> - no arguments are required and any provided argument would just be ignored.

Author: It was mainly for completeness, and yes, we can cut it short. Changing.

- if any dump fails, a fact error is returned

> ```rust

Reviewer: Maybe put a header here saying that the "keys" are sanitized and this is how they can access the values in `rhai`

Author: Added the line above
> Dumped keys (`Timeout (watchdog)`, `Timeout (msgwait)`, `Number of slots`, etc) are sanitized to simplify their access and usage in the expression language.

How does that work?

> - if dumping any of the devices fails, the fact would be a fact error

Reviewer 3: ```suggestion
- if any dump fails, the return value will be a fact error
```

Author: thanks, changed!

{
  "/path/to/a-device" => {
    "header_version"   => 2.1,
    "number_of_slots"  => 255,
    "sector_size"      => 512,
    "timeout_allocate" => 2,
    "timeout_loop"     => 1,
    "timeout_msgwait"  => 10,
    "timeout_watchdog" => 5,
    "uuid"             => "e09c8993-0cba-438d-a4c3-78e91f58ee52",
  },
  "/path/to/another-device" => {
    "header_version"   => 2.1,
    "number_of_slots"  => 255,
    "sector_size"      => 512,
    "timeout_allocate" => 2,
    "timeout_loop"     => 1,
    "timeout_msgwait"  => 10,
    "timeout_watchdog" => 5,
    "uuid"             => "e5b7c05a-1d3c-43d0-827a-9d4dd05ca54a",
  }
}
```

Specification examples:

```yaml
facts:
  - name: sbd_devices_dump
    gatherer: sbd_dump

  - name: sbd_devices_dump
    gatherer: sbd_dump

>     argument: whatever_argument # ignored

Reviewer 4: I would remove the ignored argument, as the gatherer does not accept arguments and this is something it shouldn't be present anyway
